atlas-core/src/main/java/io/atlasmap/core/AtlasMappingService.java

```diff
@@ -39,208 +43,303 @@
-			AtlasMapping atlasMapping = unmarshaller.unmarshal(streamSource, AtlasMapping.class).getValue();
-			validate(atlasMapping);
-			return atlasMapping;
-		} catch (JAXBException e) {
-			throw new AtlasValidationException(e.getMessage(), e);
-		} catch (URISyntaxException e) {
-			throw new AtlasValidationException(e.getMessage(), e);
-		}
-	}
-	
-	public void saveMappingAsFile(AtlasMapping atlasMapping, File file) throws AtlasException {
-	    saveMappingAsFile(atlasMapping, file, AtlasMappingFormat.XML);
-	}
-	
-	public void saveMappingAsFile(AtlasMapping atlasMapping, File file, AtlasMappingFormat format) throws AtlasException {
-		switch(format) {
-		case JSON: saveMappingAsJsonFile(atlasMapping, file); break;
-		case XML: saveMappingAsXmlFile(atlasMapping, file); break;
-		default: saveMappingAsXmlFile(atlasMapping, file); break;
-		}
-	}
-	
-	protected void saveMappingAsJsonFile(AtlasMapping atlasMapping, File file) throws AtlasUnsupportedException {
-	    throw new AtlasUnsupportedException("Saving as Json not implemented");
-	}
-	
-	protected void saveMappingAsXmlFile(AtlasMapping atlasMapping, File file) throws AtlasException {
-	    try {
-	        marshaller.marshal(atlasMapping ,file);
-	    } catch (JAXBException e) {
-	        throw new AtlasValidationException(e.getMessage(), e);
-	    }       
-	}
-
-	public void validate(AtlasMapping atlasMapping) throws AtlasValidationException {
-//		if(atlasMapping == null || atlasMapping.getName() == null) {
-//			throw new AtlasValidationException("AtlasMapping and name must be specified");
-//		}
-	}
-	
-	public JAXBContext getJAXBContext() {
-		return ctx;
-	}
-
-	public void setJAXBContext(JAXBContext ctx) {
-		this.ctx = ctx;
-	}
-
-	public Marshaller getMarshaller() {
-		return marshaller;
-	}
-
-	public void setMarshaller(Marshaller marshaller) {
-		this.marshaller = marshaller;
-	}
-
-	public Unmarshaller getUnmarshaller() {
-		return unmarshaller;
-	}
-
-	public void setUnmarshaller(Unmarshaller unmarshaller) {
-		this.unmarshaller = unmarshaller;
-	}
-	
-	private String stringListToColonSeparated(List<String> items) {
-		StringBuilder buffer = new StringBuilder(CONFIG_V2_PACKAGE);
-		
-		if(items == null) {
-			return null;
-		}
-		
-		if(items.isEmpty()) {
-			return buffer.toString();
-		}
-		
-		boolean first = true;
-		for (int i=0; i<items.size(); i++) {
-		    if(!CONFIG_V2_PACKAGE.equals(items.get(i))) {
-		        if(first) {
-		            buffer.append(":");
-		            first = false;
-		        }
-		        buffer.append(items.get(i));
-		        
-		        if(i < items.size()-1 ) {
-		            buffer.append(":");
-		        }
-			}
-		}
-	
-		return buffer.toString();
-	}
+    public AtlasMapping loadMapping(File file) throws AtlasValidationException {
+        return loadMapping(file, AtlasMappingFormat.XML);
+    }
+    
+    public AtlasMapping loadMapping(File file, AtlasMappingFormat format) throws AtlasValidationException {
```

Comment:
Dito. In the long run it would simplify maintenance if there is only a single method doing the callout to JAXB or Jackson (with a Reader as this as the most document oriented stream available, e.g. is a superset of inputstream as it contains the encoding)